exporter/pulsarexporter/config.go

```diff
@@ -124,3 +151,95 @@ func (cfg *Config) clientOptions() pulsar.ClientOptions {
 
 	return options
 }
+
+func (cfg *Config) getProducerOptions() (pulsar.ProducerOptions, error) {
+
+	producerOptions := pulsar.ProducerOptions{
+		Topic:                           cfg.Topic,
+		SendTimeout:                     cfg.Timeout,
+		DisableBatching:                 cfg.Producer.DisableBatching,
+		DisableBlockIfQueueFull:         cfg.Producer.DisableBlockIfQueueFull,
+		MaxPendingMessages:              cfg.Producer.MaxPendingMessages,
+		BatchingMaxPublishDelay:         cfg.Producer.BatchingMaxPublishDelay,
+		BatchingMaxSize:                 cfg.Producer.BatchingMaxSize,
+		BatchingMaxMessages:             cfg.Producer.BatchingMaxMessages,
+		PartitionsAutoDiscoveryInterval: cfg.Producer.PartitionsAutoDiscoveryInterval,
+		MaxReconnectToBroker:            cfg.Producer.MaxReconnectToBroker,
+	}
+
+	batchBuilderType, err := stringToBatchBuilderType(cfg.Producer.BatcherBuilderType)
+	if err != nil {
+		return producerOptions, err
+	}
+	producerOptions.BatcherBuilderType = batchBuilderType
+
+	compressionType, err := stringToCompressionType(cfg.Producer.CompressionType)
+	if err != nil {
+		return producerOptions, err
+	}
+	producerOptions.CompressionType = compressionType
+
+	compressionLevel, err := stringToCompressionLevel(cfg.Producer.CompressionLevel)
+	if err != nil {
+		return producerOptions, err
+	}
+	producerOptions.CompressionLevel = compressionLevel
+
+	hashingScheme, err := stringToHashingScheme(cfg.Producer.HashingScheme)
+	if err != nil {
+		return producerOptions, err
+	}
+	producerOptions.HashingScheme = hashingScheme
+
+	return producerOptions, nil
+}
+
+func stringToBatchBuilderType(builderType string) (pulsar.BatcherBuilderType, error) {
+	switch builderType {
+	case "default":
+		return pulsar.DefaultBatchBuilder, nil
+	case "key_based":
+		return pulsar.KeyBasedBatchBuilder, nil
+	default:
+		return pulsar.DefaultBatchBuilder, fmt.Errorf("producer.batchBuilderType should be one of 'default' or 'key_based'. configured value %v. Assigning default value as default", builderType)
+	}
+}
+
+func stringToCompressionType(compressionType string) (pulsar.CompressionType, error) {
```

Comment:
I think it'd be cleaner to use `encoding.TextUnmarshaler` interface instead of these functions. Look in the other components' configs. It's commonly used in the collector

Reply:
done